core/o11y/src/lib.rs

```diff
@@ -129,21 +190,27 @@ impl<'a> EnvFilterBuilder<'a> {
     }
 
     /// Construct an [`EnvFilter`] as configured.
-    pub fn finish(self) -> EnvFilter {
-        let mut env_filter = EnvFilter::new(self.rust_log);
+    pub fn finish(self) -> Result<EnvFilter, BuildEnvFilterError> {
+        let mut env_filter = EnvFilter::try_new(self.rust_log.clone())
+            .map_err(|err| BuildEnvFilterError::CreateEnvFilter(err, self.rust_log.to_string()))?;
         if let Some(module) = self.verbose {
             env_filter = env_filter
                 .add_directive("cranelift_codegen=warn".parse().expect("parse directive"))
                 .add_directive("h2=warn".parse().expect("parse directive"))
-                .add_directive("trust_dns_resolver=warn".parse().expect("parse_directive"))
-                .add_directive("trust_dns_proto=warn".parse().expect("parse_directive"));
+                .add_directive("trust_dns_resolver=warn".parse().expect("parse directive"))
+                .add_directive("trust_dns_proto=warn".parse().expect("parse directive"));
```

Comment:
By the way, am I the only one who thinks it’s dumb that `.add_directive(("module", LevelFilter::WARN).into())` or something to that effect isn’t supported?

Reply:
It is a TODO in code: https://docs.rs/tracing-subscriber/latest/src/tracing_subscriber/filter/env/directive.rs.html#13